src/trace/implementations/mod.rs

```diff
@@ -45,25 +45,30 @@ pub(crate) mod merge_batcher_col;
 
 pub use self::merge_batcher::MergeBatcher as Batcher;
 
-pub mod ord;
+// pub mod ord;
 pub mod ord_neu;
 pub mod rhh;
 
 // Opinionated takes on default spines.
-pub use self::ord::OrdValSpine as ValSpine;
-pub use self::ord::OrdKeySpine as KeySpine;
+pub use self::ord_neu::OrdValSpine as ValSpine;
+/// Alias while we work around `ord` stuff.
+pub type KeySpine<K, T, D> = ValSpine<K, (), T, D>;
+// pub use self::ord::OrdKeySpine as KeySpine;
 
 use std::ops::{Add, Sub};
 use std::convert::{TryInto, TryFrom};
+use std::borrow::{Borrow, ToOwned};
 
 use timely::container::columnation::{Columnation, TimelyStack};
 use lattice::Lattice;
 use difference::Semigroup;
 
 /// A type that names constituent update types.
 pub trait Update {
+    /// We will be able to read out references to this type, and must supply `Key::Owned` as input.
+    type Key: Ord + ToOwned<Owned = Self::KeyOwned> + ?Sized;
     /// Key by which data are grouped.
-    type Key: Ord+Clone;
+    type KeyOwned: Ord+Clone + Borrow<Self::KeyOwned>;
```

Comment:
This type is present here because it seems relatively hard to drive type inference off of `Key::Owned`, and in particular expressing constraints on that type (e.g. `Ord + Clone`). You could put a `where` clause with these constraints, but that just turned in to a where clause that all users of `Update` need to repeat as part of their constraints on `U: Update` (that it's key's offset implements certain traits).

Perhaps this can be finessed; it's certainly gross having to state the type twice, once explicitly and once as `Key::Owned`.